models/events/build.go

```diff
@@ -0,0 +1,60 @@
+package events
+
+import (
+	"time"
+
+	"github.com/gofrs/uuid"
+)
+
+type EventBuilder struct {
+	event Event
+}
+
+func NewEvent() *EventBuilder {
+	operationId, _ := uuid.NewV4()
+	return &EventBuilder{
+		event: Event{
+			CreatedAt:   time.Now(),
+			OperationID: operationId,
+		},
+	}
+}
+
+func (e *EventBuilder) ActedOn(resource uuid.UUID) *EventBuilder {
```

Comment:
ActedUpon, not "On"

Reply:
https://github.com/meshery/meshkit/pull/356/files/83dcd0952d500438cda2161e305b8ed1d12547a8#diff-5bc079c95a1811fd042d036db5a54fbae0ba3ef9e1645ce28d4eb4050b749049R24

Reply:
Why not use the same moniker for both function and attribute like what is done with Description?

You're not being consistent with the approach. In some instances you are adding a verb or preposition to the function name, but not always. Consistency facilitates legibility and comprehension by your future self and others.